pkg/handlers/downstream.go

```diff
@@ -26,24 +29,44 @@ func (h *Handler) GetDownstreamOutput(w http.ResponseWriter, r *http.Request) {
 	appSlug := mux.Vars(r)["appSlug"]
 	clusterID := mux.Vars(r)["clusterId"]
 	sequence, err := strconv.Atoi(mux.Vars(r)["sequence"])
+	output := new(types.DownstreamOutput)
 	if err != nil {
 		logger.Error(err)
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
-	a, err := store.GetStore().GetAppFromSlug(appSlug)
-	if err != nil {
-		logger.Error(err)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
+	if util.IsHelmManaged() {
+		app := helm.GetHelmApp(appSlug)
+		if app == nil {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		releaseSecret, err := helm.GetChartSecret(app.Release.Name, app.Release.Namespace, mux.Vars(r)["sequence"])
+		if err != nil {
+			logger.Error(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		if releaseSecret.Info.Status != "failed" {
```

Comment:
this needs to check for releaseSecret  != nil now

Reply:
ah of course, thanks